addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -76,66 +76,273 @@ function factory(dependencies) {
                     // uuid notification, only for (livechat) public handler
                     return;
                 }
-                if (typeof message === 'object') {
-                    switch (message.type) {
-                        case 'mail.channel_joined':
-                            return this._handleNotificationChannelJoined(message.payload);
-                    }
-                }
-                const [, model, id] = channel;
-                switch (model) {
-                    case 'ir.needaction':
-                        return this._handleNotificationNeedaction(message);
-                    case 'mail.channel':
-                        return this._handleNotificationChannel(id, message);
-                    case 'res.partner':
+                const [, , id] = channel;
+                switch (message.type) {
+                    /**
+                     * This notification trigger when new message are read inside a channel.
+                     * This is only send to the current user for multi-device synchronisation.
+                     */
+                    case 'mail.channel_seen':
+                        return this._handleNotificationChannelSeen(message.payload);
+                    /**
+                     * This notification mean that the user as fecthed the message inside his UI.
+                     * This is broadcasted to everyone in the channel.
+                     * FIXME this currently only works inside the direct message for performance raison.
```

Comment:
it's not really a fixme, we have no plan to want to support it in channels actually